tt_metal/llrt/tt_cluster.hpp

```diff
@@ -119,6 +119,20 @@ class Cluster {
     // Returns connected ethernet core on the other chip
     std::tuple<chip_id_t, CoreCoord> get_connected_ethernet_core(std::tuple<chip_id_t, CoreCoord> eth_core) const;
 
+    // Converts logical ethernet core coord to physical ethernet core coord
+    CoreCoord ethernet_core_from_logical_core(chip_id_t chip_id, const CoreCoord &logical_core) const;
+
+    // Configures routing mapping of ethernet cores
+    void initialize_routing_info_for_ethernet_cores();
+
+    // Dispatch core is managed by device, so this is an api for device to write the routing info to each eth core used
+    // in FD tunneling Returns logical eth core that communicates with specified dispatch core
+    CoreCoord get_and_configure_corresponding_eth_core_for_device(
+        chip_id_t chip_id, CoreCoord physical_dispatch_core, EthRouterMode mode, chip_id_t connected_chip_id) const;
+
+    // Writes eth routing config to device, assumes that device is initialized
+    void set_internal_routing_info_for_ethernet_cores(bool enable_internal_routing) const;
```

Comment:
can you add comment here explaining what internal routing is

Reply:
Will do